app/dcr_monitor/imagebuilder/kaniko.go

```diff
@@ -155,33 +156,57 @@ func (b *KanikoImageBuilder) deleteJob(name string) error {
 	return nil
 }
 
-func (b *KanikoImageBuilder) BuildImage(j *db.Job, bucket string, baseImage string, image string) error {
+func (b *KanikoImageBuilder) BuildImage(j *db.Job, baseImage string, image string) error {
 
 	buildArgs := []string{
 		fmt.Sprintf("--context=%s", j.BuildContextPath),
 		fmt.Sprintf("--destination=%s", image),
-		// TODO: this should be signed URL (See https://github.com/manatee-project/manatee/issues/23)
-		fmt.Sprintf("--build-arg=OUTPUTPATH=%s", fmt.Sprintf("gs://%s/%s/output/out-%s-%s", bucket, j.Creator, j.UUID, j.JupyterFileName)),
+		fmt.Sprintf("--build-arg=OUTPUT_SIGNED_URL=%s", j.OutputPutSignedUrl),
 		fmt.Sprintf("--build-arg=JUPYTER_FILENAME=%s", j.JupyterFileName),
 		fmt.Sprintf("--build-arg=USER_WORKSPACE=%s", fmt.Sprintf("%s-workspace", j.Creator)),
 		fmt.Sprintf("--build-arg=BASE_IMAGE=%s", baseImage),
-		// TODO: this should be signed URL (See https://github.com/manatee-project/manatee/issues/23)
-		fmt.Sprintf("--build-arg=CUSTOMTOKEN_CLOUDSTORAGE_PATH=%s", fmt.Sprintf("gs://%s/%s/output/%s-token", bucket, j.Creator, j.UUID)),
+		fmt.Sprintf("--build-arg=CUSTOMTOKEN_SIGNED_URL=%s", j.CustomTokenPutSignedUrl),
+	}
+	var envs []corev1.EnvVar
+
+	if os.Getenv("STORAGE_TYPE") == "MINIO" {
+		envs = append(envs, corev1.EnvVar{
+			Name:  "AWS_ACCESS_KEY_ID",
+			Value: os.Getenv("AWS_ACCESS_KEY_ID"),
+		},
+			corev1.EnvVar{
+				Name:  "AWS_SECRET_ACCESS_KEY",
+				Value: os.Getenv("AWS_SECRET_ACCESS_KEY"),
+			},
+			corev1.EnvVar{
+				Name:  "S3_ENDPOINT",
+				Value: fmt.Sprintf("http://%s", os.Getenv("S3_ENDPOINT")),
+			},
+			corev1.EnvVar{
+				Name:  "AWS_REGION",
+				Value: "us-east-1",
+			},
+			corev1.EnvVar{
+				Name:  "S3_FORCE_PATH_STYLE",
+				Value: "true",
+			},
+		)
```

Comment:
The kaniko build context only support file url like s3://xx or gs:/xxx. A signed url is not allowed

Reply:
are you sure? the code seems like to support it though..

https://github.com/GoogleContainerTools/kaniko/blob/e328007bc1fa0d8c2eacf1918bebbabc923abafa/pkg/buildcontext/buildcontext.go#L61
https://github.com/GoogleContainerTools/kaniko/blob/main/pkg/buildcontext/https.go

Reply:
You are right, https is supported. But the signed url for minikube deployment is http by default. It's a little complex to enable tls to support https. So I think we can leave it for future feature.

Reply:
You are right, https is supported. But the signed url for minikube deployment is http by default. It's a little complex to enable tls to support https. So I think we can leave it for future feature.

Reply:
The main issue with this code is that the `ImageBuilder` needs to be aware of all supported storage types, and have this kind of logic for handling that. This is not a good practice because it makes code more challenging to understand. i.e., if someone wants to add AWS S3 storage, they should be aware that not only `Storage` but also `ImageBuidler` will be affected. If this is strictly required, that means we need to redefine `ImageBuilder` such that it includes `Storage` by itself.

A better way, in my opinion, is to add these extra variables a part of `Job`, not a part of `ImageBuilder` itself. e.g., add extra envs for image building in `Job` record, and then have kaniko builder just add those variables to the container.

Reply:
nit: please don't resolve the comments if it's not certainly resolved!